src/main/resources/config/application.yml

```diff
@@ -0,0 +1,67 @@
+# All env variables that are used in one place
+# Exception: TERRA_ENV (which sets the active profile), which is parsed before this is.
+env:
+  db:
+    host: jdbc:postgresql://${DATABASE_HOSTNAME:127.0.0.1}:5432
+    init: ${INIT_DB:false}
+    stairway:
+      name: ${STAIRWAY_DATABASE_NAME:stairwaylib}
+      pass: ${STAIRWAY_DATABASE_USER_PASSWORD:stairwaypwd}
+      user: ${STAIRWAY_DATABASE_USER:stairwayuser}
+    ws:
+      name: ${STAIRWAY_DATABASE_NAME:wm}
+      pass: ${STAIRWAY_DATABASE_USER_PASSWORD:wmpwd}
+      user: ${STAIRWAY_DATABASE_USER:wmuser}
+  gcp-sa-path: ${CLOUD_TRACE_SA_PATH:rendered/service-account.json}
+
+logging.pattern.level: '%X{requestId} %5p'
+
+server:
+  compression:
+    enabled: true
+    mime-types: text/css,application/javascript
+  port: 8080
+
+spring:
+  cloud.gcp:
+    credentials.location: file:${env.gcp-sa-path}
+    trace.enabled: true
+  profiles.active: ${TERRA_ENV:dev} # This is evaluated before the "env" section above
+  resources:
+    cache:
+      cachecontrol:
+        max-age: 0
+        must-revalidate: true
+    static-locations: classpath:/api/
+  sleuth.sampler.probability: 0.1
+
+workspace:
+  datarepo.instances:
```

Comment:
I think it's important for the main application.yml to have entries for all possible application-specific configs. This file doesn't have an entry for workspace.datarepo.instances.terra, but that value does show up in the env-specific overrides. Two questions / suggestions:

1) For this PR, can you add an entry for workspace.datarepo.instances.terra = 'jade.datarepo-dev'? This would follow your pattern of "the base application.yml has values we use for local testing against dev".

2) Would there be a longer-term plan to move entirely away from having all of these values hard-coded in the main application.yml, and have each WSM server only have a single TDR server it's configured to point to?

Reply:
The goal is to remove the datarepo.instances stanza from the main config entirely, and have it in the profiles only; this is an in-between state that would be resolved by the end of the chain of PR's that this would be part of.

Reply:
The end state for this is the same as the current state of `workspace.sam.basePath`